lib/elixir_google_scraper_web/router.ex

```diff
@@ -16,10 +17,25 @@ defmodule ElixirGoogleScraperWeb.Router do
   # coveralls-ignore-start
   pipeline :api do
     plug(:accepts, ["json"])
+  end
 
-    scope "/api/v1", ElixirGoogleScraperWeb, as: :api_v1 do
-      post("/oauth/token", Api.V1.TokenController, :create)
-    end
+  pipeline :api_auth do
+    plug(ExOauth2Provider.Plug.VerifyHeader, otp_app: :elixir_google_scraper, realm: "Bearer")
+
+    plug(ExOauth2Provider.Plug.EnsureAuthenticated,
+      handler: ElixirGoogleScraperWeb.Api.ErrorHandler
+    )
+
+    plug(:set_current_user)
+  end
+
+  scope "/api/v1", ElixirGoogleScraperWeb, as: :api_v1 do
+    pipe_through(:api)
+
+    post("/oauth/token", Api.V1.TokenController, :create)
+
+    pipe_through(:api_auth)
```

Comment:
Oh, I don't know that we could do it like this 😆 

I usually create a different scope like this

```elixir
scope "/api/v1", ElixirGoogleScraperWeb, as: :api_v1 do
  pipe_through(:api)

  post("/oauth/token", Api.V1.TokenController, :create)
end

scope "/api/v1", ElixirGoogleScraperWeb, as: :api_v1 do
  pipe_through(:api_auth)

  resources("/keywords", Api.V1.KeywordController, only: [:create])
end
```